apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/QuestionCard.tsx

```diff
@@ -98,6 +98,23 @@ export default function QuestionCard({
   const question = localSurvey.questions[questionIdx];
   const open = activeQuestionId === question.id;
   const [openAdvanced, setOpenAdvanced] = useState(question.logic && question.logic.length > 0);
+  const [questionsWithEmptyLabel, setquestionsWithEmptyLabel] = useState<string[]>([]);
+
+  const editNextButtonLabel = (labelValue: string) => {
+    localSurvey.questions.forEach((q, index) => {
+      if (!q.buttonLabel || q.buttonLabel?.trim() === "" || questionsWithEmptyLabel.includes(q.id)) {
+        if (!questionsWithEmptyLabel.includes(q.id)) {
+          setquestionsWithEmptyLabel((prevArray) => [...prevArray, q.id]);
+        }
+        updateQuestion(index, { buttonLabel: labelValue });
+      } else {
+        updateQuestion(questionIdx, { buttonLabel: labelValue });
+      }
+    });
+  };
+  useEffect(() => {
+    setquestionsWithEmptyLabel([]);
```

Comment:
I'm not getting why we need the `questionsWithEmptyLabel` state. Why isn't it working without it?

Reply:
Actually this state was used to update next button value of a question, which has been already set
But this was required on with if we check for `emptyNextButtonLabel` with `onChange`
`onBlur` makes it much simpler 😊💪